test/common/performance/aggregator/aggregator.go

```diff
@@ -259,22 +242,46 @@ func eventsToTimestampsArray(events *map[string]*timestamp.Timestamp) []time.Tim
 		t, _ := ptypes.Timestamp(v)
 		values = append(values, t)
 	}
-	sort.Slice(values, func(x, y int) bool { return values[x].Before(values[y]) })
 	return values
 }
 
-func publishThpt(timestamps []time.Time, q *quickstore.Quickstore, metricName string) error {
-	for i, t := range timestamps[1:] {
-		var thpt uint
-		j := i - 1
-		for j >= 0 && t.Sub(timestamps[j]) <= time.Second {
-			thpt++
-			j--
+func publishThpt(q *quickstore.Quickstore,
```

Comment:
I haven't fully understood this function yet, but one early question: why remove the sorting?

Reply:
It's simply because we don't need it :)

What the original algorithm does is, calculate the RPS/QPS at each request and add them as a sample point, which is unnecessary and sometimes making the plot less intuitive. And the time complexity is high - O(m*n), where m is the total number of requests, n is the average RPS/QPS.

The new algorithm uses `t.Unix()` to round the timestamp to second, and calculate the RPS/QPS only once per second into a map, with the time complexity to be O(m), space complexity to be O(number of seconds). And also for every second, it has both success and error metrics, so that the plots will look more consecutive.

Below are more comparisons for these two approaches:
IMC Channel:
- Before: https://mako.dev/run?run_key=5988673954775040&~pl=1&~pe=1&~st=1&~dl=1&~de=1&~dt=1&~pet=1&~det=1
- After:  https://mako.dev/run?run_key=5755730302861312&~pl=1&~pe=1&~st=1&~dl=1&~de=1&~dt=1&~pet=1&~det=1

IMC Broker:
- Before: https://mako.dev/run?run_key=5646700605603840&~pl=1&~pe=1&~st=1&~dl=1&~de=1&~dt=1&~pet=1&~det=1
- After: https://mako.dev/run?run_key=6326108060385280&~pl=1&~pe=1&~st=1&~dl=1&~de=1&~dt=1&~pet=1&~det=1

Reply:
Thanks, sounds reasonable!